plugins/modules/win_dhcp_server.ps1

```diff
@@ -0,0 +1,259 @@
+#!powershell
+# Copyright: (c) 2021 Sebastian Gruber (@sgruber94) ,dacoso GmbH All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#AnsibleRequires -CSharpUtil Ansible.Basic
+$spec = @{
+    options             = @{
+        type              = @{ type = "str"; choices = "scope", "exclusion"; required =$true }
+        version           = @{ type = "str"; choices = "IPv4", "IPv6"; default = "IPv4" }
+        scope             = @{ type = "str"; aliases = "prefix" }
+        name              = @{ type = "str"; required =$true  }
+        validlifetime     = @{ type = "str" ; default = "0.04:00:00" }
+        preferredlifetime = @{ type = "str" ; default = "0.02:00:00" }
+        startrange        = @{ type = "str" }
+        endrange          = @{ type = "str" }
+        subnetmask        = @{ type = "str" }
+        state             = @{ type = "str"; choices = "absent", "present"; default = "present" }
+        computername      = @{ type = "str" }
```

Comment:
This is probably better to name `dhcp_server` as `computername` is a big vague. We do the same thing for the AD modules where the `-ComputerName` equivalent for those modules is mapped to the `domain_server` module option.